### DISCUSSION TITLE
create-next-app add type-check command for TypeScript projects

### DISCUSSION THREAD

**Discussion by VladimirMikulic** [vercel/next.js#41145] [Ideas]
### Describe the feature you'd like to request

When creating TypeScript based Next.js project, it would be helpful if TS type checking command was included by default:
```js
scripts: {
    // ...
    "type-check": "tsc --pretty --noEmit"
}
```

### Describe the solution you'd like

```js
scripts: {
    // ...
    "type-check": "tsc --pretty --noEmit"
}
```

### Describe alternatives you've considered

Currently, I'm adding this `type-check` manually to my projects but it would be great if it was included by default for TypeScript projects.

**Comment by ijjk** [comment]
Hi, is there a specific reason this is needed versus the built-in type checking done during `next build`?

**Reply by VladimirMikulic** [reply]
Thanks for a fast reply @ijjk.
`next-build` builds the whole project so I need to watch for TypeScript step and click `CTRL + C` to exit the process.
Running `next-build` (after `next dev`) locally also corrupts sometimes `.next` folder which gets app in a weird state so the only solution is to remove `.next` folder and then run `next dev` again.
